mgmt/rest/fixtures/mock_metric_manager.go

```diff
@@ -0,0 +1,302 @@
+
+Copyright 2016 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fixtures
+
+import (
+	"errors"
+	"time"
+
+	"github.com/intelsdi-x/snap/control/plugin/cpolicy"
+	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/core/serror"
+)
+
+//////MockLoadedPlugin/////
+
+type MockLoadedPlugin struct {
+	MyName    string
+	MyType    string
+	MyVersion int
+}
+
+func (m MockLoadedPlugin) Name() string       { return m.MyName }
+func (m MockLoadedPlugin) TypeName() string   { return m.MyType }
+func (m MockLoadedPlugin) Version() int       { return m.MyVersion }
+func (m MockLoadedPlugin) Plugin() string     { return "" }
+func (m MockLoadedPlugin) IsSigned() bool     { return false }
+func (m MockLoadedPlugin) Status() string     { return "" }
+func (m MockLoadedPlugin) PluginPath() string { return "" }
+func (m MockLoadedPlugin) LoadedTimestamp() *time.Time {
+	t := time.Date(2016, time.September, 6, 0, 0, 0, 0, time.UTC)
+	return &t
+}
+func (m MockLoadedPlugin) Policy() *cpolicy.ConfigPolicy { return cpolicy.New() }
+func (m MockLoadedPlugin) HitCount() int                 { return 0 }
+func (m MockLoadedPlugin) LastHit() time.Time            { return time.Now() }
+func (m MockLoadedPlugin) ID() uint32                    { return 0 }
+
+var pluginCatalog []core.CatalogedPlugin = []core.CatalogedPlugin{
+	MockLoadedPlugin{MyName: "foo", MyType: "collector", MyVersion: 2},
+	MockLoadedPlugin{MyName: "bar", MyType: "publisher", MyVersion: 3},
+	MockLoadedPlugin{MyName: "foo", MyType: "collector", MyVersion: 4},
+	MockLoadedPlugin{MyName: "baz", MyType: "publisher", MyVersion: 5},
+	MockLoadedPlugin{MyName: "foo", MyType: "processor", MyVersion: 6},
+	MockLoadedPlugin{MyName: "foobar", MyType: "processor", MyVersion: 1},
+}
+
+//////MockCatalogedMetric/////
+
+type MockCatalogedMetric struct{}
+
+func (m MockCatalogedMetric) Namespace() core.Namespace {
+	return core.NewNamespace("one", "two", "three")
+}
+func (m MockCatalogedMetric) Version() int                      { return 5 }
+func (m MockCatalogedMetric) LastAdvertisedTime() time.Time     { return time.Time{} }
+func (m MockCatalogedMetric) Policy() *cpolicy.ConfigPolicyNode { return cpolicy.NewPolicyNode() }
+func (m MockCatalogedMetric) Description() string               { return "This Is A Description" }
+func (m MockCatalogedMetric) Unit() string                      { return "" }
+
+var metricCatalog []core.CatalogedMetric = []core.CatalogedMetric{
```

Comment:
Would it make sense to define pluginCatalog and this in close proximity so that they are easier to find while reading through it?